### PR TITLE
refactor(iso-script): improve flexibility and add new images

### PR DESCRIPTION
**Pull request by HryshcIlya**
- Add associative array for image configurations
- Include new images: aurora, cosmic, server-zfs
- Dynamically generate desktop options
- Simplify logic for adding parameters to image name
- Improve handling of specific desktop options (dx, asus, surface)
- Enhance code modularity for easier future modifications
- Update VARIANT selection logic for all images

BREAKING CHANGE: The script now uses a new configuration structure for image options, which may require updates to any external dependencies or documentation referencing the old script structure. New images have been added and VARIANT selection has been modified for existing images.

### CODE REVIEW

**Comment by RoyalOughtness** [comment]
Thanks for this, I will review this later today

File: generate_secureblue_iso.sh [hunk 1]
```diff
@@ -12,97 +13,76 @@ function is_yes {
     esac
 }
 
-desktop_options=("kinoite" "cinnamon" "bluefin" "silverblue" "sericea" "wayblue-wayfire" "wayblue-sway" "wayblue-river" "wayblue-hyprland")
-desktop_options_asus=("silverblue" "kinoite")
+# Define image configurations
+declare -A image_configs=(
+    ["server"]="Server"
+    ["server-zfs"]="Server"
+    ["silverblue"]="Silverblue"
```
Comment (RoyalOughtness) [inline generate_secureblue_iso.sh]: silverblue also has asus options

File: generate_secureblue_iso.sh [hunk 2]
```diff
@@ -12,97 +13,76 @@ function is_yes {
...
+    ["kinoite"]="Kinoite"
+    ["bluefin"]="Silverblue:dx"
+    ["sericea"]="Sericea"
+    ["wayblue-wayfire"]="Sericea"
+    ["wayblue-sway"]="Sericea"
```
Comment (RoyalOughtness) [inline generate_secureblue_iso.sh:25]: instead of "sericea", shouldn't this just be the empty string? same for cinnamon and others

**Review by RoyalOughtness — CHANGES_REQUESTED** [review]
looks good, just a few comments

File: generate_secureblue_iso.sh (hunk 1 above)
Comment (HryshcIlya) [reply generate_secureblue_iso.sh]: Yes, indeed, I overlooked the `asus` option for Silverblue and Kinoite. It has been fixed now. Apologies for any delays in communication; we might be in different time zones, leading to asynchronous interactions.

File: generate_secureblue_iso.sh (hunk 2 above)
Comment (HryshcIlya) [reply generate_secureblue_iso.sh:25]: For images like `wayblue-sway`, `bluefin`, and others, I selected the most similar variants based on my understanding of their intended functionality. While exploring the [documentation](https://github.com/JasonN3/build-container-installer), I discovered that if no variant is specified, it defaults to using the `server` variant. This default behavior would require users to manually partition the disk, which can be quite inconvenient.

Comment (RoyalOughtness) [reply generate_secureblue_iso.sh:25]: @HryshcIlya makes sense, thanks for clarifying!

File: generate_secureblue_iso.sh (hunk 1 above)
Comment (RoyalOughtness) [reply generate_secureblue_iso.sh]: @HryshcIlya no worries! it's FOSS, there's no rush :smile:

**Pull request merged**